packages/ckeditor5-link/src/linkui.ts

```diff
@@ -487,30 +552,43 @@ export default class LinkUI extends Plugin {
 			const linkCommand: LinkCommand = editor.commands.get( 'link' )!;
 			const t = locale.t;
 
+			button.bind( 'isEnabled' ).to( linkCommand, 'value', href => !!href );
+
 			button.bind( 'href' ).to( linkCommand, 'value', href => {
 				return href && ensureSafeUrl( href, allowedProtocols );
 			} );
 
-			button.bind( 'label' ).to( linkCommand, 'value', href => {
+			button.bind( 'label' ).to( linkCommand, 'value', this, 'selectedLinksProviderLink', ( href, provider ) => {
+				if ( provider ) {
+					return provider.item.label;
+				}
+
 				if ( !href ) {
 					return t( 'This link has no URL' );
 				}
 
-				return isScrollableToTarget( editor, href ) ? href.slice( 1 ) : href;
+				return href;
 			} );
 
-			button.bind( 'icon' ).to( linkCommand, 'value', href => {
-				return href && isScrollableToTarget( editor, href ) ? icons.bookmarkSmall : undefined;
+			button.bind( 'icon' ).to( this, 'selectedLinksProviderLink', provider => {
+				return provider && provider.item.icon;
 			} );
 
-			button.bind( 'isEnabled' ).to( linkCommand, 'value', href => !!href );
+			button.bind( 'tooltip' ).to( this, 'selectedLinksProviderLink', provider => {
```

Comment:
Also here. As we search for the same href in multiple binds maybe we could watch for changes of linkCommand value and update all fields at once?